src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -817,6 +826,9 @@ void Thread::HijackReturnAddressWorker(StackFrameIterator* frameIterator, Hijack
 #endif
 
         *ppvRetAddrLocation = (void*)pfnHijackFunction;
+#if defined(TARGET_ARM64)
+        *ppvRetAddrLocation = PacSignPtr(*ppvRetAddrLocation);
+#endif // TARGET_ARM64
```

Comment:
```suggestion
        void* pvHIjackAddr = (void*)pfnHijackFunction;
#if defined(TARGET_ARM64)
        pvHIjackAddr = PacSignPtr(pvHIjackAddr);
#endif // TARGET_ARM64
        *ppvRetAddrLocation = pvHIjackAddr;
```
We should avoid writing the wrong value first, and then overwriting it with the correct value. It can cause interesting race conditions.

Reply:
Done